R/pkg/R/schema.R

```diff
@@ -162,7 +162,7 @@ checkType <- function(type) {
             },
             m = {
               # Map type
-              m <- regexec("^map<(.+),(.+)>$", type)
+              m <- regexec("map<(string|character),(.+)>", type)
```

Comment:
This is just a legacy sanity check since R type parsing is now delegated into SQL parser. We should actually remove this method entirely (see https://github.com/apache/spark/commit/70f1bcd7bcd42b30eabcf06a9639363f1ca4b449).

Can you file a JIRA, review https://spark.apache.org/contributing.html closely, and update this PR to remove this with a set of tests?

Reply:
I think this PR is irrelevant if type parsing is to be delegated into SQL parser. I am removing it.

Reply:
You can just remove this checkType though.